src/mount/mount.ceph.c

```diff
@@ -425,24 +425,41 @@ static void ceph_mount_info_free(struct ceph_mount_info *cmi)
 	free(cmi->cmi_conf);
 }
 
-static int finalize_options(struct ceph_mount_info *cmi)
+static int append_secret_option(struct ceph_mount_info *cmi)
 {
-	int pos;
+	int pos = strlen(cmi->cmi_opts);
+	const char *key_name = cmi->cmi_name;
+	const char *secret = cmi->cmi_secret;
 
-	if (cmi->cmi_secret[0] || is_kernel_secret(cmi->cmi_name)) {
-		int ret;
-		char secret_option[SECRET_OPTION_BUFSIZE];
-
-		ret = get_secret_option(cmi->cmi_secret, cmi->cmi_name,
-					secret_option, sizeof(secret_option));
-		if (ret < 0)
-			return ret;
+	if (!secret[0] && !is_kernel_secret(cmi->cmi_name))
+		return 0;
 
-		pos = strlen(cmi->cmi_opts);
-		if (pos)
-			pos = safe_cat(&cmi->cmi_opts, &cmi->cmi_opts_len, pos, ",");
-		pos = safe_cat(&cmi->cmi_opts, &cmi->cmi_opts_len, pos, secret_option);
+	if (pos)
+		pos = safe_cat(&cmi->cmi_opts, &cmi->cmi_opts_len, pos, ",");
+
+	/* when parsing kernel options (-o remount) we get '<hidden>' as the secret */
+	if (secret[0] && (strcmp(secret, "<hidden>") != 0)) {
+		int ret = set_kernel_secret(secret, key_name);
+		if (ret < 0) {
+			if (ret == -ENODEV || ret == -ENOSYS) {
+				/* old kernel; fall back to secret= in options */
+				pos = safe_cat(&cmi->cmi_opts,
+					       &cmi->cmi_opts_len, pos,
+					       "secret=");
+				pos = safe_cat(&cmi->cmi_opts,
+					       &cmi->cmi_opts_len, pos, secret);
```

Comment:
Nit: perhaps write it as
```
		if (ret < 0) {
			if (ret != -ENODEV && ret != -ENOSYS) {
				fprintf(...);
				return ret;
			}

			/* old kernel; fall back to secret= in options */
			...
			return 0;
		}
```
to avoid else after return and reduce indentation?

Reply:
Sounds good. I fell prey to copypasta.